diff --git a/Makefile.am b/Makefile.am
index 26081d8..e8ff9de 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -100,7 +100,7 @@ src_connmand_SOURCES = $(gdbus_sources) $(gdhcp_sources) $(gweb_sources) \
 			src/inotify.c
 
 src_connmand_LDADD = $(builtin_libadd) @GLIB_LIBS@ @DBUS_LIBS@ \
-				@XTABLES_LIBS@ @GNUTLS_LIBS@ -lresolv -ldl -lrt
+				@XTABLES_LIBS@ @GNUTLS_LIBS@ -ldl -lrt
 
 src_connmand_LDFLAGS = -Wl,--export-dynamic \
 				-Wl,--version-script=$(srcdir)/src/connman.ver
@@ -127,7 +127,7 @@ vpn_connman_vpnd_SOURCES = $(gdbus_sources) $(builtin_vpn_sources) \
 			vpn/vpn-agent.c vpn/vpn-agent.h
 
 vpn_connman_vpnd_LDADD = $(builtin_vpn_libadd) @GLIB_LIBS@ @DBUS_LIBS@ \
-				@GNUTLS_LIBS@ -lresolv -ldl
+				@GNUTLS_LIBS@ -ldl
 
 vpn_connman_vpnd_LDFLAGS = -Wl,--export-dynamic \
 				-Wl,--version-script=$(srcdir)/vpn/vpn.ver
@@ -239,7 +239,7 @@ if WISPR
 noinst_PROGRAMS += tools/wispr
 
 tools_wispr_SOURCES = $(gweb_sources) tools/wispr.c
-tools_wispr_LDADD = @GLIB_LIBS@ @GNUTLS_LIBS@ -lresolv
+tools_wispr_LDADD = @GLIB_LIBS@ @GNUTLS_LIBS@ 
 endif
 
 if TOOLS
@@ -257,13 +257,13 @@ tools_supplicant_test_SOURCES = $(gdbus_sources) tools/supplicant-test.c \
 tools_supplicant_test_LDADD = @GLIB_LIBS@ @DBUS_LIBS@
 
 tools_web_test_SOURCES = $(gweb_sources) tools/web-test.c
-tools_web_test_LDADD = @GLIB_LIBS@ @GNUTLS_LIBS@ -lresolv
+tools_web_test_LDADD = @GLIB_LIBS@ @GNUTLS_LIBS@ 
 
 tools_resolv_test_SOURCES = gweb/gresolv.h gweb/gresolv.c tools/resolv-test.c
-tools_resolv_test_LDADD = @GLIB_LIBS@ -lresolv
+tools_resolv_test_LDADD = @GLIB_LIBS@ 
 
 tools_wpad_test_SOURCES = gweb/gresolv.h gweb/gresolv.c tools/wpad-test.c
-tools_wpad_test_LDADD = @GLIB_LIBS@ -lresolv
+tools_wpad_test_LDADD = @GLIB_LIBS@ 
 
 tools_stats_tool_LDADD = @GLIB_LIBS@
 
diff --git a/configure.ac b/configure.ac
index 764a453..2ce68d1 100644
--- a/configure.ac
+++ b/configure.ac
@@ -173,10 +173,10 @@ AM_CONDITIONAL(PPTP_BUILTIN, test "${enable_pptp}" = "builtin")
 
 AC_CHECK_HEADERS(resolv.h, dummy=yes,
 	AC_MSG_ERROR(resolver header files are required))
-AC_CHECK_LIB(resolv, ns_initparse, dummy=yes, [
-	AC_CHECK_LIB(resolv, __ns_initparse, dummy=yes,
-		AC_MSG_ERROR(resolver library support is required))
-])
+#AC_CHECK_LIB(resolv, ns_initparse, dummy=yes, [
+#	AC_CHECK_LIB(resolv, __ns_initparse, dummy=yes,
+#		AC_MSG_ERROR(resolver library support is required))
+#])
 
 AC_CHECK_FUNC(signalfd, dummy=yes,
 			AC_MSG_ERROR(signalfd support is required))
