diff --git a/src/via_id.c b/src/via_id.c
index fe88114..f3d3124 100644
--- a/src/via_id.c
+++ b/src/via_id.c
@@ -145,6 +145,7 @@ static struct ViaCardIdStruct ViaCardId[] = {
     {"ECS C7VCM",                             VIA_VM800,   0x1019, 0xAA2D, VIA_DEVICE_CRT},
     {"PCChips V21G",                          VIA_VM800,   0x1019, 0xAA51, VIA_DEVICE_CRT},
     {"Hewlett Packard DX2020",                VIA_VM800,   0x103C, 0x3027, VIA_DEVICE_CRT},
+    {"HITACHI FLORA Se210 RK1",               VIA_VM800,   0x1043, 0x13E2, VIA_DEVICE_CRT | VIA_DEVICE_LCD},
     {"Asustek P5VDC-MX",                      VIA_VM800,   0x1043, 0x3344, VIA_DEVICE_CRT},
     {"Asustek P5VDC-TVM",                     VIA_VM800,   0x1043, 0x81CE, VIA_DEVICE_CRT},
     {"Foxconn P4M800P7MB-RS2H",               VIA_VM800,   0x105B, 0x0CF0, VIA_DEVICE_CRT},
