From 06d06fdd2d45350f144ff3d79353e817b30811a7 Mon Sep 17 00:00:00 2001
From: Vic Lee <llyzs@163.com>
Date: Thu, 23 May 2013 13:35:24 -0700
Subject: [PATCH] core/info: server missing audio info flags.

---
 libfreerdp/core/info.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/libfreerdp/core/info.c b/libfreerdp/core/info.c
index a792d23..cf74d1d 100644
--- a/libfreerdp/core/info.c
+++ b/libfreerdp/core/info.c
@@ -240,6 +240,8 @@ BOOL rdp_read_info_packet(wStream* s, rdpSettings* settings)
 	Stream_Seek_UINT32(s); /* CodePage */
 	Stream_Read_UINT32(s, flags); /* flags */
 
+	settings->AudioCapture = ((flags & RNS_INFO_AUDIOCAPTURE) ? TRUE : FALSE);
+	settings->AudioPlayback = ((flags & INFO_NOAUDIOPLAYBACK) ? FALSE : TRUE);
 	settings->AutoLogonEnabled = ((flags & INFO_AUTOLOGON) ? TRUE : FALSE);
 	settings->RemoteApplicationMode = ((flags & INFO_RAIL) ? TRUE : FALSE);
 	settings->RemoteConsoleAudio = ((flags & INFO_REMOTECONSOLEAUDIO) ? TRUE : FALSE);
-- 
1.8.1.6

