diff --git a/channels/drive/client/drive_file.c b/channels/drive/client/drive_file.c
index 4d4e6e8..9abbc64 100644
--- a/channels/drive/client/drive_file.c
+++ b/channels/drive/client/drive_file.c
@@ -436,7 +436,11 @@ BOOL drive_file_set_information(DRIVE_FILE* file, UINT32 FsInformationClass, UIN
 	int status;
 	char* fullpath;
 	struct STAT st;
+#ifndef __UCLIBC__
 	struct timeval tv[2];
+#else
+	struct timespec tv[2];
+#endif
 	UINT64 LastWriteTime;
 	UINT32 FileAttributes;
 	UINT32 FileNameLength;
@@ -457,13 +461,24 @@ BOOL drive_file_set_information(DRIVE_FILE* file, UINT32 FsInformationClass, UIN
 				return FALSE;
 
 			tv[0].tv_sec = st.st_atime;
+#ifndef __UCLIBC__
 			tv[0].tv_usec = 0;
+#else
+			tv[0].tv_nsec = 0;
+#endif
 			tv[1].tv_sec = (LastWriteTime > 0 ? FILE_TIME_RDP_TO_SYSTEM(LastWriteTime) : st.st_mtime);
+#ifndef __UCLIBC__
 			tv[1].tv_usec = 0;
+#else
+			tv[0].tv_nsec = 0;
+#endif
 #ifndef WIN32
-/* TODO on win32 */                        
+/* TODO on win32 */      
+#ifndef __UCLIBC__                  
                         futimes(file->fd, tv);
-
+#else
+			futimens(file->fd, tv);
+#endif
 			if (FileAttributes > 0)
 			{
 				m = st.st_mode;
