--- polkit-0.104/src/polkitbackend/polkitbackendlocalauthorizationstore.c	2012-01-09 16:50:27.573369343 -0500
+++ polkit-0.104/src/polkitbackend/polkitbackendlocalauthorizationstore.c	2012-01-09 16:34:25.749095408 -0500
@@ -725,6 +725,7 @@
             break;
         }
 
+#if !defined(__UCLIBC__)
       /* if no identity specs matched and identity is a user, match against netgroups */
       if (ll == NULL && POLKIT_IS_UNIX_USER (identity))
         {
@@ -739,6 +740,7 @@
                 break;
             }
         }
+#endif
 
       if (ll == NULL)
         continue;
--- polkit-0.104/src/polkitbackend/polkitbackendlocalauthority.c	2012-01-09 16:50:41.703369211 -0500
+++ polkit-0.104/src/polkitbackend/polkitbackendlocalauthority.c	2012-01-09 16:34:11.324619300 -0500
@@ -698,6 +698,7 @@
   GList *ret;
 
   ret = NULL;
+#if !defined(__UCLIBC__)
   name = polkit_unix_netgroup_get_name (POLKIT_UNIX_NETGROUP (group));
 
   if (setnetgrent (name) == 0)
@@ -739,6 +740,7 @@
 
  out:
   endnetgrent ();
+#endif
   return ret;
 }
 
