diff -Nuar webkit-1.8.3.orig/Source/JavaScriptCore/wtf/Assertions.cpp webkit-1.8.3/Source/JavaScriptCore/wtf/Assertions.cpp
--- webkit-1.8.3.orig/Source/JavaScriptCore/wtf/Assertions.cpp	2012-08-20 16:26:04.000000000 +0000
+++ webkit-1.8.3/Source/JavaScriptCore/wtf/Assertions.cpp	2012-11-04 00:59:49.415465313 +0000
@@ -52,7 +52,7 @@
 #include <windows.h>
 #endif
 
-#if OS(DARWIN) || OS(LINUX)
+#if OS(DARWIN) || OS(LINUX) && !defined(__UCLIBC__)
 #include <cxxabi.h>
 #include <dlfcn.h>
 #include <execinfo.h>
@@ -249,7 +249,7 @@
 
 void WTFGetBacktrace(void** stack, int* size)
 {
-#if OS(DARWIN) || OS(LINUX)
+#if OS(DARWIN) || OS(LINUX) && !defined(__UCLIBC__)
     *size = backtrace(stack, *size);
 #elif OS(WINDOWS) && !OS(WINCE)
     // The CaptureStackBackTrace function is available in XP, but it is not defined
