diff --git a/channels/drive/client/drive_file.c b/channels/drive/client/drive_file.c
index e6b2062..39cffdb 100644
--- a/channels/drive/client/drive_file.c
+++ b/channels/drive/client/drive_file.c
@@ -462,7 +462,7 @@ BOOL drive_file_set_information(DRIVE_FILE* file, UINT32 FsInformationClass, UIN
 			tv[1].tv_usec = 0;
 #ifndef WIN32
 /* TODO on win32 */                        
-#ifdef ANDROID
+#ifdef __UCLIBC__
 			utimes(file->fullpath, tv);
 #else
 			futimes(file->fd, tv);
