Index: xserver-xorg-video-savage/src/savage_driver.c
===================================================================
--- xserver-xorg-video-savage.orig/src/savage_driver.c	2008-05-22 07:53:05.000000000 +0200
+++ xserver-xorg-video-savage/src/savage_driver.c	2008-05-22 20:55:51.000000000 +0200
@@ -1292,7 +1292,7 @@
      * We support bpp of 8, 16, and 32.
      */
 
-    if (!xf86SetDepthBpp(pScrn, 0, 0, 0, Support32bppFb))
+    if (!xf86SetDepthBpp(pScrn, 16, 0, 0, Support32bppFb))
 	return FALSE;
     else {
         int requiredBpp;
