diff -Naur libssh-0.5.2.orig/ConfigureChecks.cmake libssh-0.5.2/ConfigureChecks.cmake
--- libssh-0.5.2.orig/ConfigureChecks.cmake	2011-09-19 11:39:14.000000000 -0400
+++ libssh-0.5.2/ConfigureChecks.cmake	2011-09-19 11:39:22.000000000 -0400
@@ -45,6 +45,13 @@
 check_include_file(pty.h HAVE_PTY_H)
 check_include_file(termios.h HAVE_TERMIOS_H)
 
+if (HAVE_ARGP_H)
+  check_symbol_exists(__UCLIBC__ "features.h" HAVE_UCLIBC)
+  if (HAVE_UCLIBC)
+    set(CMAKE_REQUIRED_LIBRARIES ${CMAKE_REQUIRED_LIBRARIES} argp)
+  endif (HAVE_UCLIBC)
+endif (HAVE_ARGP_H)
+
 if (WIN32)
   check_include_file(wspiapi.h HAVE_WSPIAPI_H)
   if (NOT HAVE_WSPIAPI_H)
