diff -Naur uClibc-0.9.33.2.orig/libpthread/nptl/sysdeps/pthread/Makefile.in uClibc-0.9.33.2/libpthread/nptl/sysdeps/pthread/Makefile.in
--- uClibc-0.9.33.2.orig/libpthread/nptl/sysdeps/pthread/Makefile.in	2012-05-15 12:16:59.000000000 -0400
+++ uClibc-0.9.33.2/libpthread/nptl/sysdeps/pthread/Makefile.in	2012-05-15 12:19:25.000000000 -0400
@@ -80,7 +80,7 @@
 $(libpthread_pthread_OUT)/pt-crtn.o: $(libpthread_pthread_OUT)/pt-crtn.S
 	$(compile.S)
 
-$(libpthread_pthread_OUT)/pt-initfini.s: $(PTHREAD_INITFINI) | $(top_builddir)include/bits/uClibc_config.h
+$(libpthread_pthread_OUT)/pt-initfini.s: $(PTHREAD_INITFINI) | $(top_builddir)include/bits/uClibc_config.h $(top_builddir)/include/bits/uClibc_locale_data.h
 	$(compile.c)
 $(libpthread_pthread_OUT)/defs.h: $(PTHREAD_INITFINI)
 	$(do_sed) -n -e '/@TESTS_BEGIN/,/@TESTS_END/p' $< | \
