--- libssh-0.5.2/examples/CMakeLists.txt	2012-03-23 16:37:48.673736468 -0400
+++ libssh-0.5.2/examples/CMakeLists.txt	2012-03-23 16:36:48.807279268 -0400
@@ -42,7 +42,7 @@
 target_link_libraries(senddata ${LIBSSH_SHARED_LIBRARY})
 
 add_executable(libsshpp libsshpp.cpp)
-target_link_libraries(libsshpp ${LIBSSH_SHARED_LIBRARY})
+target_link_libraries(libsshpp ${LIBSSH_SHARED_LIBRARY} "-pthread")
 
 add_executable(libsshpp_noexcept libsshpp_noexcept.cpp)
-target_link_libraries(libsshpp_noexcept ${LIBSSH_SHARED_LIBRARY})
+target_link_libraries(libsshpp_noexcept ${LIBSSH_SHARED_LIBRARY} "-pthread")
