diff --git a/client_examples/Makefile.am b/client_examples/Makefile.am
index 2779d53..0030a57 100644
--- a/client_examples/Makefile.am
+++ b/client_examples/Makefile.am
@@ -14,19 +14,19 @@ vnc2mpg_LDADD=$(LDADD) $(FFMPEG_HOME)/libavformat/libavformat.a $(FFMPEG_HOME)/l
 FFMPEG_CLIENT=vnc2mpg
 endif
 
-if HAVE_LIBSDL
-SDLVIEWER=SDLvncviewer
+# if HAVE_LIBSDL
+# SDLVIEWER=SDLvncviewer
 
-SDLvncviewer_CFLAGS=$(SDL_CFLAGS)
-SDLvncviewer_SOURCES=SDLvncviewer.c scrap.c scrap.h
+# SDLvncviewer_CFLAGS=$(SDL_CFLAGS)
+# SDLvncviewer_SOURCES=SDLvncviewer.c scrap.c scrap.h
 
-if HAVE_X11
-X11_LIB=-lX11
-endif
+# if HAVE_X11
+# X11_LIB=-lX11
+# endif
 
 # thanks to autoconf, this looks ugly
-SDLvncviewer_LDADD=$(LDADD) $(SDL_LIBS) $(X11_LIB)
-endif
+# SDLvncviewer_LDADD=$(LDADD) $(SDL_LIBS) $(X11_LIB)
+# endif
 
 if HAVE_LIBGTK
 GTKVIEWER=gtkvncviewer
@@ -36,7 +36,7 @@ gtkvncviewer_LDADD=$(LDADD) $(GTK_LIBS)
 endif
 
 
-noinst_PROGRAMS=ppmtest $(SDLVIEWER) $(GTKVIEWER) $(FFMPEG_CLIENT) backchannel
+noinst_PROGRAMS=ppmtest $(GTKVIEWER) $(FFMPEG_CLIENT) backchannel
 
 
 
